packages/server/test/integration/cypress_spec.coffee

```diff
@@ -336,6 +336,28 @@ describe "lib/cypress", ->
         expect(browsers.open).to.be.calledWithMatch(ELECTRON_BROWSER, {url: "http://localhost:8888/__/#/tests/integration/test2.coffee"})
         @expectExitWith(0)
 
+    it "runs project by limiting spec files via config.testFiles string glob pattern", ->
+      cypress.start(["--run-project=#{@todosPath}", "--config=testFiles=#{@todosPath}/tests/test2.coffee"])
+      .then =>
+        expect(browsers.open).to.be.calledWithMatch(ELECTRON_BROWSER, {url: "http://localhost:8888/__/#/tests/integration/test2.coffee"})
+        @expectExitWith(0)
+
+    it "runs project by limiting spec files via config.testFiles as an array of string glob patterns", ->
+      cypress.start(["--run-project=#{@todosPath}", "--config=testFiles=**/test2.coffee,**/test1.js"])
+      .then =>
+        expect(browsers.open).to.be.calledWithMatch(ELECTRON_BROWSER, {url: "http://localhost:8888/__/#/tests/integration/test2.coffee"})
+      .then =>
+        expect(browsers.open).to.be.calledWithMatch(ELECTRON_BROWSER, {url: "http://localhost:8888/__/#/tests/integration/test1.js"})
+        @expectExitWith(0)
+
+    it "runs project by limiting spec files via config.testFiles as an array of string glob patterns", ->
+      cypress.start(["--run-project=#{@todosPath}", "--config='[\"testFiles=**/test2.coffee\",\"**/test1.js\"]'"])
```

Comment:
what is this syntax `--config='["testFiles=`? there is a `[` after `--config=`, which I do not see anywhere else in the tests

Reply:
I was not sure what the syntax was supposed to be, but found a similar multi-value configuration option, `blacklistHosts` [here](https://github.com/cypress-io/cypress/blob/92b91fe514e5ff6286b4d4e26d2df23062bdf869/packages/server/test/unit/args_spec.coffee#L140-L148), so I followed that pattern. In the link, it appears that the multi-value value of blacklistHosts is simply JSON.stringified as the CLI option.

If this is not correct, we can change it. I may have found or looked in the wrong places to find the syntax.

The output of this discussion will also impact https://github.com/cypress-io/cypress-documentation/pull/2184

Reply:
I think the consensus is to follow the same syntax as `blacklistHosts`; in which case this is the correct syntax: `--config=testFiles=["glob1","glob2"]`.

Note that `spec` and `tag` are run with a slightly different syntax, so we may want to reconsider the syntax across all multi-value values in the future.

Reply:
notice in that example you've linked, the `[` does not included in the args in the cli

Reply:
In hindsight, I don't think my example and how it was conveyed was very clear. Unless I am misunderstanding you, I think the CLI args are: something like this:

`--config=pageLoadTimeout=10000,waitForAnimations=false,hosts={"foobar.com":"127.0.0.1"},blacklistHosts=["one.com","www.two.io"]`

My reasoning is outlined, hopefully more clearly, below:

- `blacklistHosts` is an array [here](https://github.com/cypress-io/cypress/blob/92b91fe514e5ff6286b4d4e26d2df23062bdf869/packages/server/test/unit/args_spec.coffee#L128)
- it is turned into JSON [here](https://github.com/cypress-io/cypress/blob/92b91fe514e5ff6286b4d4e26d2df23062bdf869/packages/server/test/unit/args_spec.coffee#L140)
- that JSON array string is used for the blacklistHosts CLI command [here](https://github.com/cypress-io/cypress/blob/92b91fe514e5ff6286b4d4e26d2df23062bdf869/packages/server/test/unit/args_spec.coffee#L148)
- the `join()` is joining all the CLI arguments; one of which is blacklistHosts which is `blacklistHosts=["one.com", "www.two.io"]`

Reply:
Per our discussion, I understand what you are saying now. Resolving this comment and updating code accordingly.